packages/gatsby-plugin-fastify/src/plugins/decorators.ts

```diff
@@ -3,15 +3,22 @@ import fp from "fastify-plugin";
 import { moduleHeaderDecorator, setHeaderDecorator } from "../utils/headers";
 
 import type { FastifyPluginAsync } from "fastify";
+import { Page } from "../gatsby/header-builder";
 
 declare module "fastify" {
   interface FastifyReply {
     appendModuleHeader: typeof moduleHeaderDecorator;
     setHeader: FastifyReply["header"];
+    mode: Page["mode"] | "";
```

Comment:
Would this be the same/better?

```suggestion
    mode: Page["mode"] | string;
```

Reply:
I don't think so. That would allow any string value vs the actual `modes` + `""`. Ideally, it would just be `Page["mode"]` - the only reason I added `| ""` was because Fastify docs recommend instantiating decorators with the same type they are intended to use, ie: `""` vs `null | undefined`.

Reply:
ah, right. That makes sense.